naga/src/back/spv/block.rs

```diff
@@ -2034,14 +2034,33 @@ impl<'w> BlockContext<'w> {
         }
     }
 
+    /// Generate one or more SPIR-V blocks for `naga_block`.
+    ///
+    /// Use `label_id` as the label for the SPIR-V entry point block.
+    ///
+    /// If control reaches the end of the SPIR-V block, terminate it according
+    /// to `exit`.
+    ///
+    /// If the block contains [`Break`] or [`Continue`] statements,
+    /// `loop_context` supplies the labels of the SPIR-V blocks to jump to. If
+    /// either of these labels are `None`, then it should have been a Naga
+    /// validation error for the corresponding statement to occur in this
+    /// context.
+    ///
+    /// Return `true` if control can reach the end of the block. This means that
+    /// `exit` was used. Otherwise, return `false`, indicating that `exit` was
+    /// unused.
+    ///
+    /// [`Break`]: Statement::Break
+    /// [`Continue`]: Statement::Continue
     pub(super) fn write_block(
         &mut self,
         label_id: Word,
         naga_block: &crate::Block,
         exit: BlockExit,
         loop_context: LoopContext,
         debug_info: Option<&DebugInfoInner>,
-    ) -> Result<(), Error> {
```

Comment:
nitpick(non-blocking): I think it's difficult to keep in mind what this `bool` means while reading code outside of the signature.

suggestion: Let's use a one-off `enum`:

```rust
enum ControlReachesEndOfBlock {
    Yes,
    No,
}
```

Reply:
Okay. I'd considered doing that. `bool` is usually not great in these situations.